### PR TITLE
fix(VRipple): add keyboard trigger support

### PR DESCRIPTION
**Pull request by fpdrozd**
## Description

This PR makes it possible to trigger the ripple effect by pressing the space or enter key on the focused element.

## Motivation and Context
#10285

## How Has This Been Tested?
unit | visually

## Markup:
<details>

```vue
<template>
  <v-container>
    <v-btn color="primary">Primary</v-btn>
  </v-container>
</template>

<script>
export default {
  data: () => ({
    //
  })
}
</script>
```
</details>

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and breaking changes).
- [x] My code follows the code style of this project.
- [ ] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)
- [ ] I've added new examples to the kitchen (applies to new features and breaking changes in core library)

### CODE REVIEW

**Comment by fpdrozd** [comment]
@johnleider Hi, I am a beginner contributor and I tried to write these unit tests. Am I missing something? Can I correct them or should someone more experienced take care of that?

**Comment by johnleider** [comment]
Just need markup (Playground) so this can be tested. Thank you!

File: packages/vuetify/src/directives/ripple/index.ts [hunk 1]
```diff
@@ -163,6 +171,18 @@ function rippleHide (e: Event) {
   ripples.hide(element)
 }
 
+let keyboardRipple = false
+function keyboardRippleShow (e: KeyboardEvent) {
+  if (!keyboardRipple && (e.keyCode === 13 || e.keyCode === 32)) {
```
Comment (johnleider) [inline packages/vuetify/src/directives/ripple/index.ts]: Should compare against the `keyCodes` object from `util/helpers.ts`

File: packages/vuetify/src/directives/ripple/index.ts [hunk 2]
```diff
@@ -19,15 +19,23 @@ export interface RippleOptions {
...
+  return e.constructor.name === 'KeyboardEvent'
+}
+
+const calculate = (e: MouseEvent | TouchEvent | KeyboardEvent, el: HTMLElement, value: RippleOptions = {}) => {
+  let localX = 0
+  let localY = 0
+  if (!isKeyboardEvent(e)) {
+    const offset = el.getBoundingClientRect()
+    const target = isTouchEvent(e) ? e.touches[e.touches.length - 1] : e
+    localX = target.clientX - offset.left
```
Comment (johnleider) [inline packages/vuetify/src/directives/ripple/index.ts]: blank line between

File: packages/vuetify/src/directives/ripple/index.ts [hunk 3]
```diff
@@ -19,15 +19,23 @@ export interface RippleOptions {
...
-  const localX = target.clientX - offset.left
-  const localY = target.clientY - offset.top
+function isKeyboardEvent (e: MouseEvent | TouchEvent | KeyboardEvent): e is KeyboardEvent {
+  return e.constructor.name === 'KeyboardEvent'
+}
+
+const calculate = (e: MouseEvent | TouchEvent | KeyboardEvent, el: HTMLElement, value: RippleOptions = {}) => {
+  let localX = 0
+  let localY = 0
+  if (!isKeyboardEvent(e)) {
```
Comment (johnleider) [inline packages/vuetify/src/directives/ripple/index.ts]: blank line between

File: packages/vuetify/src/directives/ripple/index.ts [hunk 4]
```diff
@@ -19,15 +19,23 @@ export interface RippleOptions {
...
-const calculate = (e: MouseEvent | TouchEvent, el: HTMLElement, value: RippleOptions = {}) => {
-  const offset = el.getBoundingClientRect()
-  const target = isTouchEvent(e) ? e.touches[e.touches.length - 1] : e
-  const localX = target.clientX - offset.left
-  const localY = target.clientY - offset.top
+function isKeyboardEvent (e: MouseEvent | TouchEvent | KeyboardEvent): e is KeyboardEvent {
+  return e.constructor.name === 'KeyboardEvent'
+}
+
+const calculate = (e: MouseEvent | TouchEvent | KeyboardEvent, el: HTMLElement, value: RippleOptions = {}) => {
```
Comment (johnleider) [inline packages/vuetify/src/directives/ripple/index.ts]: for readability break these into new lines for each prop

File: packages/vuetify/src/directives/ripple/index.ts [hunk 5]
```diff
@@ -50,7 +58,7 @@ const calculate = (e: MouseEvent | TouchEvent, el: HTMLElement, value: RippleOpt
 
 const ripples = {
   /* eslint-disable max-statements */
-  show (e: MouseEvent | TouchEvent, el: HTMLElement, value: RippleOptions = {}) {
+  show (e: MouseEvent | TouchEvent | KeyboardEvent, el: HTMLElement, value: RippleOptions = {}) {
```
Comment (johnleider) [inline packages/vuetify/src/directives/ripple/index.ts]: for readability break these into new lines for each prop

**Comment by fpdrozd** [comment]
@jacekkarczmarczyk @KaelWD Would you guys please review this PR?

**Comment by johnleider** [comment]
Requesting **Final** comments from @vuetifyjs/core-team

File: packages/vuetify/src/directives/ripple/__tests__/ripple.spec.ts [hunk 6]
```diff
@@ -66,4 +74,55 @@ describe('ripple.ts', () => {
     wrapper.setProps({ ripple: true })
     expect(div.element['_ripple'].enabled).toBe(true)
   })
+
+  it('Ripple should be triggered by the mousedown event', () => {
+    jest.useFakeTimers()
```
Comment (sh7dm) [inline packages/vuetify/src/directives/ripple/__tests__/ripple.spec.ts:79]: Move this to `beforeEach` and add `afterEach` with `jest.useRealTimers`

File: packages/vuetify/src/directives/ripple/index.ts [hunk 7]
```diff
@@ -1,9 +1,13 @@
 // Styles
 import './VRipple.sass'
 
-import { VNode, VNodeDirective } from 'vue'
+// Utilities
+import { keyCodes } from '../../util/helpers'
```
Comment (johnleider) [inline packages/vuetify/src/directives/ripple/index.ts:5]: Swap with line 6

**Pull request merged**